Modules/mathmodule.c

```
@@ -840,6 +840,20 @@ math_lcm_impl(PyObject *module, PyObject * const *args,
 }
 
 
+static void
+set_exc_value(double x)
+{
+    PyObject *exc = PyErr_GetRaisedException();
+    PyObject *value = PyFloat_FromDouble(x);
+
+    if (value) {
+        PyObject_SetAttrString(exc, "value", value);
```

This call can fail. Should we do something about it? or should we simply ignore the exception? or maybe chain it?

---

> or should we simply ignore the exception?

I don't see better option so far.  (Ditto for cmath.)

BTW, maybe I should also drop ``if (value) {`` clause.

> or maybe chain it?

I doubt it's appropriate here.  We don't handle ValueError here.  This code just alter the raised exception object.

---

> BTW, maybe I should also drop if (value) { clause.

I don't think you should, otherwise `PyObject_SetAttr` will be called with an exception set.

---

> will be called with an exception set

Why it's bad in this case?

> and with a NULL value it's the same as deleting the attribute

But we are already ignore errors from the PyObject_SetAttrString :-)

---

> Why it's bad in this case?

Don't we have an assert() somewhere that would fail if the error indicator is set?

---

I failed to find such case.

Though, we could explicitly call (twice) PyErr_Occurred() in appropriate places.

Looking on other cases of using PyErr_GetRaisedException() - I think that unraisable exceptions should be used here.

---

See cmath's pr, I did such change: https://github.com/python/cpython/pull/134995.  Does it make sense for you?